src/core/search.test.ts

```diff
@@ -45,4 +47,10 @@ describe("ticket search", () => {
       errors: [error0, error1].map(serialize),
     });
   });
+
+  it("searches jira cloud before jira server", () => {
+    expect(stdadapters.indexOf(JiraCloud)).toBeLessThan(
+      stdadapters.indexOf(JiraServer)
+    );
+  });
```

Comment:
I have a question and a remark:

1. Why does the order matter here?
2. Even though the adapters are listed in this order, it doesn't mean the search with the cloud adapter will wait for the first to complete before trying the next one. If you look at `src/core/search.ts` (`search` function) you'll see that we `map` across all adapters and then `await` with `Promise.all` – meaning the adapter scans are run concurrently.

Reply:
If you want to guarantee the cloud adapter scan completes before attempting the Jira Server scan, you need to either change the `search` behavior` or do it within one adapter which is why I proposed this in the issue:
https://github.com/bitcrowd/tickety-tick/issues/349#issuecomment-1274536884

If you're sure they'll never both complete successfully for any given page, you don't need to care about the order. Assuming one succeeds, but the other fails, errors will be discarded and the successful result used. Ideally though, you should fail as early as possible and avoid unnecessary network requests if you can identify which kind of Jira you're on.

Reply:
Thanks for jumping onto this (again) @pmeinhardt 💙 

From a first glance, you're of course absolutely right: nothing keeps the `jira-server` adapater to be ready with results **before** the `jira-cloud` one on cloud instances. The `jira` ID on the body tag is there for both types of installations.

It's probably more future-proof to handle that in one adapter. This PR should for now have fixed "not working on Jira server" for now. I'll open an issue for a cleanup to merge the two adapters. I'd personally prefer it over changing the behavior of the `search` function.

See https://github.com/bitcrowd/tickety-tick/issues/359